src/run.ts

```diff
@@ -17,223 +25,419 @@ const deployedServices = [
-  await runner.run_command_and_output(
-    "db yarn",
-    ["yarn", "install"],
-    "services/database"
+async function run_fe_locally(runner: LabeledProcessRunner) {
+  const apiUrl = await getCloudFormationStackOutputValue(
+    "mfp-localstack",
+    "ApiUrl"
   );
+  await writeLocalUiEnvFile(apiUrl!);
+  runner.run_command_and_output("ui", ["npm", "start"], "services/ui-src");
+}
+
+async function run_cdk_watch(
+  runner: LabeledProcessRunner,
+  options: { stage: string }
+) {
   await runner.run_command_and_output(
-    "db svls",
-    ["serverless", "dynamodb", "install", "--stage", "local"],
-    "services/database"
-  );
-  runner.run_command_and_output(
-    "db",
+    "CDK watch",
     [
-      "serverless",
-      "offline",
-      "start",
-      "--stage",
-      "local",
-      "--lambdaPort",
-      "3003",
+      "yarn",
+      "cdk",
+      "watch",
+      "--context",
+      `stage=${options.stage}`,
+      "--no-rollback",
     ],
-    "services/database"
+    "."
   );
 }
 
-// run_api_locally uses the serverless-offline plugin to run the api lambdas locally
-async function run_api_locally(runner: LabeledProcessRunner) {
+function isColimaRunning() {
+  try {
+    const output = execSync("colima status 2>&1", {
+      encoding: "utf-8",
+      stdio: "pipe",
+    }).trim();
+    return output.includes("running");
+  } catch {
+    return false;
+  }
+}
+
+function isLocalStackRunning() {
+  try {
+    return execSync("localstack status", {
+      encoding: "utf-8",
+      stdio: "pipe",
+    }).includes("running");
+  } catch {
+    return false;
+  }
+}
+
+async function run_watch(options: { stage: string }) {
+  const runner = new LabeledProcessRunner();
+  await prepare_services(runner);
+
+  run_cdk_watch(runner, options);
+  run_fe_locally(runner);
+}
+
+async function getCloudFormationStackOutputValue(
+  stackName: string,
+  outputName: string
+) {
+  const cloudFormationClient = new CloudFormationClient({ region });
+  const command = new DescribeStacksCommand({ StackName: stackName });
+  const response = await cloudFormationClient.send(command);
+  return response.Stacks?.[0]?.Outputs?.find(
+    (output) => output.OutputKey === outputName
+  )?.OutputValue;
+}
+
+async function run_local() {
+  const runner = new LabeledProcessRunner();
+  await prepare_services(runner);
+
+  if (!isColimaRunning()) {
+    throw "Colima needs to be running.";
+  }
+
+  if (!isLocalStackRunning()) {
+    throw "LocalStack needs to be running.";
+  }
+
+  process.env.AWS_DEFAULT_REGION = "us-east-1";
+  process.env.AWS_ACCESS_KEY_ID = "localstack";
+  process.env.AWS_SECRET_ACCESS_KEY = "localstack"; // pragma: allowlist secret
+  process.env.AWS_ENDPOINT_URL = "http://localhost:4566";
+
   await runner.run_command_and_output(
-    "api deps",
-    ["yarn", "install"],
-    "services/app-api"
+    "CDK local bootstrap",
+    [
+      "yarn",
+      "cdklocal",
+      "bootstrap",
+      "aws://000000000000/us-east-1",
```

Comment:
non-blocking
nit: could we make this a variable?
`"aws://000000000000/us-east-1"`
will it ever change? at least provide a comment?